src/resources/assets/assetsApi.ts

```diff
@@ -90,39 +87,127 @@ export class AssetsAPI {
+      instance,
+      path,
+      map,
+      transformResponse,
+      assetChunker
+    );
+    this.list = generateListEndpoint(
+      instance,
+      path,
+      map,
+      true,
+      transformResponse
+    );
+    this.retrieve = generateRetrieveEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
+    this.update = generateUpdateEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
+    this.search = generateSearchEndpoint(
+      instance,
+      path,
+      map,
+      transformResponse
+    );
     this.delete = generateDeleteEndpointWithParams(instance, path, map);
   }
+
+  public async retrieveSubtree(id: types.CogniteInternalId, depth: number) {
+    const currentDepth: number = 0;
+    const rootAssetList = await this.retrieve([{ id }]);
+    return this.getAssetSubtree(rootAssetList, currentDepth, depth);
+  }
+
+  private transformToAssetListObject = (assets: types.Asset[]) => {
+    const assetArray = assets.map(asset => new Asset(this.client, asset));
+    return new AssetList(this.client, assetArray);
+  };
+
+  private async getAssetSubtree(
+    assets: AssetList,
+    currentDepth: number,
+    depth: number = Infinity
+  ): Promise<AssetList> {
+    const subtree: AssetList = assets;
+    if (depth > currentDepth) {
+      const children = await this.getChildren(assets);
+      if (children.length !== 0) {
+        const subtreeOfChildren = await this.getAssetSubtree(
+          children,
+          currentDepth + 1,
+          depth
+        );
+        subtree.push(...subtreeOfChildren);
+      }
+    }
+    return subtree;
+  }
+
+  private getChildren = async (assets: AssetList) => {
+    const ids = assets.map(asset => asset.id);
+    const chunks = chunk(ids, 100);
+    const assetsArray: Asset[] = [];
+    for (const chunkOfAssetIds of chunks) {
+      const childrenList = await this.client.assets
+        .list({
+          filter: {
+            parentIds: chunkOfAssetIds,
+          },
+        })
+        .autoPagingToArray({ limit: Infinity });
+      assetsArray.push(...childrenList);
+    }
+    return new AssetList(this.client, assetsArray);
+  };
 }
 
 export type AssetCreateEndpoint = (
-  items: ExternalAssetItem[]
-) => Promise<Asset[]>;
+  items: types.ExternalAssetItem[]
+) => Promise<AssetList>;
 
 export type AssetListEndpoint = (
-  scope?: AssetListScope
+  scope?: types.AssetListScope
 ) => CursorAndAsyncIterator<Asset>;
 
-export type AssetRetrieveEndpoint = (ids: AssetIdEither[]) => Promise<Asset[]>;
+export type AssetRetrieveEndpoint = (
+  ids: types.AssetIdEither[]
+) => Promise<AssetList>;
 
-export type AssetUpdateEndpoint = (changes: AssetChange[]) => Promise<Asset[]>;
+export type AssetUpdateEndpoint = (
+  changes: types.AssetChange[]
+) => Promise<AssetList>;
 
 export type AssetSearchEndpoint = (
-  query: AssetSearchFilter
-) => Promise<Asset[]>;
+  query: types.AssetSearchFilter
```

Comment:
you can import all types from types.ts, but do `Asset as AssetProps`, like
```
import { ExternalAssetItem, AssetListScope, AssetIdEither, AssetChange, AssetSearchFilter, CogniteInternalId, Asset as TypeAsset } from '../../types/types';
```
and only asset class separately
```
import { Asset } from '../classes/asset';
```

and then
```
... query: AssetSearchFilter
```
```
private transformToAssetListObject = (assets: AssetProps[]) => {
```

So you don't have ugly `types.Something`  everywhere

Reply:
you can come up with better name then `AssetProps` ofc